<a href="https://colab.research.google.com/github/electiva1995-2026/datamining/blob/main/Ejercicio3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1. Archivo requerido para el ejercicio

El archivo solicitado por el ejercicio es Ejercicio 3.csv. Este archivo debe contener una muestra de 547 individuos con tres atributos principales: Weight, Cholesterol y Gender.

Columna	Descripción	Tipo esperado
Weight	Peso de la persona, medido en libras.	Numérico entero
Cholesterol	Nivel de colesterol de la persona.	Numérico entero
Gender	Género codificado como 0 y 1.	Numérico entero


2. Revisión inicial del archivo

In [55]:
import pandas as pd

df = pd.read_csv("https://raw.githubusercontent.com/electiva1995-2026/datamining/refs/heads/main/Ejercicio%203.csv")

df.head()


,Weight,Cholesterol,Gender
0,102,111,1
1,115,135,1
2,115,136,1
3,140,167,0
4,130,158,1


3. Validación de dimensiones

In [56]:
df.shape

(547, 3)

4. Revisión de tipos de datos

In [57]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 547 entries, 0 to 546
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   Weight       547 non-null    int64
 1   Cholesterol  547 non-null    int64
 2   Gender       547 non-null    int64
dtypes: int64(3)
memory usage: 12.9 KB


5. Revisión de valores nulos

In [58]:
df.isnull().sum()

,0
Weight,0
Cholesterol,0
Gender,0


6. Revisión de registros duplicados

In [59]:
df.duplicated().sum()

np.int64(57)

7. Estadística descriptiva

In [60]:
df.describe()

,Weight,Cholesterol,Gender
count,547.000000,547.000000,547.000000
mean,143.572212,170.433272,0.513711
std,30.837275,39.147189,0.500269
min,95.000000,102.000000,0.000000
25%,116.000000,136.000000,0.000000
50%,140.000000,169.000000,1.000000
75%,171.000000,208.000000,1.000000
max,203.000000,235.000000,1.000000


8. Análisis preliminar de variables

8.1 Peso - Weight
•	El peso promedio es de aproximadamente 143.57 libras.
•	El peso mínimo registrado es de 95 libras.
•	El peso máximo registrado es de 203 libras.
•	La variable presenta suficiente variabilidad para contribuir a la formación de grupos.

8.2 Colesterol - Cholesterol
•	El colesterol promedio es de aproximadamente 170.43.
•	El valor mínimo registrado es de 102.
•	El valor máximo registrado es de 235.
•	Esta variable es clave para identificar posibles grupos de riesgo relacionados con salud cardiovascular.

8.3 Género - Gender
•	La media de género es de 0.51.
•	Como la codificación es 0 = Mujer y 1 = Hombre, la muestra se encuentra bastante equilibrada.
•	El género puede complementar la interpretación de los grupos, aunque debe analizarse con cuidado porque está codificado como variable binaria.


9. Distribución de género

In [61]:
df["Gender"].value_counts()

,count
Gender,
1,281
0,266


10. Revisión de valores atípicos

In [63]:
variables_numericas = ["Weight", "Cholesterol"]

for columna in variables_numericas:
    media = df[columna].mean()
    desviacion = df[columna].std()

    limite_inferior = media - 2 * desviacion
    limite_superior = media + 2 * desviacion

    atipicos = df[
        (df[columna] < limite_inferior) |
        (df[columna] > limite_superior)
    ]

    print("Variable:", columna)
    print("Media:", round(media, 2))
    print("Desviación estándar:", round(desviacion, 2))
    print("Límite inferior:", round(limite_inferior, 2))
    print("Límite superior:", round(limite_superior, 2))
    print("Cantidad de valores atípicos:", len(atipicos))
    print("-" * 40)


Variable: Weight
Media: 143.57
Desviación estándar: 30.84
Límite inferior: 81.9
Límite superior: 205.25
Cantidad de valores atípicos: 0
----------------------------------------
Variable: Cholesterol
Media: 170.43
Desviación estándar: 39.15
Límite inferior: 92.14
Límite superior: 248.73
Cantidad de valores atípicos: 0
----------------------------------------


10.1 Interpretación para Weight
Para la variable Weight, serían considerados atípicos los pesos menores a 81.90 libras o mayores a 205.25 libras. En el archivo analizado, el peso mínimo es 95 y el máximo es 203, por lo tanto no se identifican valores atípicos.


10.2 Interpretación para Cholesterol
Para la variable Cholesterol, serían considerados atípicos los valores menores a 92.14 o mayores a 248.73. En el archivo analizado, el valor mínimo es 102 y el máximo es 235, por lo tanto no se identifican valores atípicos.


11. Código completo recomendado